composites/Plugin/SnippetPreview/components/SnippetPreview.js

```diff
@@ -197,41 +199,52 @@ const Amp = styled.div`
  * Highlights a keyword with strong React elements.
  *
  * @param {string} locale ISO 639 (2/3 characters) locale.
- * @param {string} keyword The keyword.
+ * @param {string[]} keywordForms The array of keyword forms.
  * @param {string} text The text in which to highlight a keyword.
  * @param {string} cleanText Optional. The text in which to highlight a keyword
  *                           without special characters and diacritics.
  *
  * @returns {ReactElement} React elements to be rendered.
  */
-function highlightKeyword( locale, keyword, text, cleanText ) {
-	if ( ! keyword ) {
+function highlightKeyword( locale, keywordForms, text, cleanText ) {
+	if ( keywordForms.length === 0 ) {
 		return text;
 	}
 
-	/*
-	 * When a text has been cleaned up from special characters and diacritics
-	 * we need to match against a cleaned up keyword as well.
-	 */
+	keywordForms = uniq( flatten( keywordForms ) );
+
+
+	// Clean the text from special characters and diacritics.
 	const textToUse = cleanText ? cleanText : text;
-	const keywordToUse = cleanText ? replaceSpecialCharactersAndDiacritics( keyword ) : keyword;
 
-	// Match keyword case-insensitively.
-	const keywordMatcher = createWordRegex( keywordToUse, "", false );
+	// Initiate an array of transliterated forms.
+	const keywordFormsCleaned = [];
+	const keywordFormsTransliterated = [];
 
-	text = textToUse.replace( keywordMatcher, function( matchedKeyword ) {
-		return `{{strong}}${ matchedKeyword }{{/strong}}`;
+	keywordForms.forEach( function( form ) {
+		/*
+	    * When a text has been cleaned up from special characters and diacritics
+	    * we need to match against a cleaned up keyword as well.
+	    */
+		form = cleanText ? replaceSpecialCharactersAndDiacritics( form ) : form;
+
+		keywordFormsCleaned.push( form );
+
+		// Transliterate the keyword for highlighting
+		const formTransliterated = transliterate( form, locale );
+
+		if ( formTransliterated !== form ) {
+			keywordFormsTransliterated.push( formTransliterated );
+		}
 	} );
 
-	// Transliterate the keyword for highlighting
-	const transliteratedKeyword = transliterate( keyword, locale );
-	if ( transliteratedKeyword !== keyword ) {
-		const transliteratedKeywordMatcher = createWordRegex( transliteratedKeyword, "", false );
-		// Let the transliteration run on the text with no previous replacements.
-		text = text.replace( transliteratedKeywordMatcher, function( matchedKeyword ) {
-			return `{{strong}}${ matchedKeyword }{{/strong}}`;
-		} );
-	}
+	const allKeywordForms = keywordFormsCleaned.concat( keywordFormsTransliterated );
+
+	const keywordFormsMatcher = createRegexFromArray( allKeywordForms, false, "", false );
+
+	text = textToUse.replace( keywordFormsMatcher, function( matchedKeyword ) {
```

Comment:
This is overwriting the variable passed to this function. I would say as a general rule, do not do this as it could have unexpected consequences.